### PR TITLE
chore(headless SSR): create commerce SSR sub-package

### PR DESCRIPTION
**Pull request by y-lakhdar**
Add boilerplate to for the ssr commerce samples

https://coveord.atlassian.net/browse/KIT-3398

### CODE REVIEW

File: packages/headless/src/ssr-commerce.index.ts [hunk 1]
```diff
@@ -0,0 +1,222 @@
+import {polyfillCryptoNode} from './api/analytics/analytics-crypto-polyfill';
+
+polyfillCryptoNode();
```
Comment (louis-bompart) [inline packages/headless/src/ssr-commerce.index.ts:3]: Scrap that.
We're gonna put the bar at the right height from the get-go on this package
(i.e. flush those darn polyfill 😂 )

**Review by louis-bompart — APPROVED** [review]
Nice!

File: packages/headless/src/ssr-commerce.index.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+  facetResponseSelectedValuesSelector,
+} from './features/facets/facet-set/facet-set-selectors';
+
+export {
+  currentPageSelector,
+  maxPageSelector,
+  currentPagesSelector,
+} from './features/pagination/pagination-selectors';
+
+//#region Grouped actions
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:94]: We have commerce-specific actions, I think we should export those instead.

File: packages/headless/src/ssr-commerce.index.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+export type {
+  DateFacet,
+  DateFacetState,
+} from './controllers/commerce/core/facets/date/headless-commerce-date-facet';
+export type {RegularFacetValue} from './controllers/commerce/core/facets/headless-core-commerce-facet';
+export type {
+  NumericFacet,
+  NumericFacetState,
+} from './controllers/commerce/core/facets/numeric/headless-commerce-numeric-facet';
+export type {RegularFacet} from './controllers/commerce/core/facets/regular/headless-commerce-regular-facet';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:74]: Why are we exporting those in particular?

I think we could start with the bare minimum exports and add them as we implement the ssr controllers.

File: packages/headless/src/ssr-commerce.index.ts [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+export * from './features/question-answering/question-answering-actions-loader';
+export * from './features/breadcrumb/breadcrumb-actions-loader';
+export * from './features/recent-queries/recent-queries-actions-loader';
+export * from './features/recent-results/recent-results-actions-loader';
+export * from './features/excerpt-length/excerpt-length-actions-loader';
+export * from './features/result-preview/result-preview-actions-loader';
+export * from './features/generated-answer/generated-answer-actions-loader';
+export {buildResultTemplatesManager} from './features/result-templates/result-templates-manager';
+//#endregion
+// Analytics Actions
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:132]: Not useful in commerce

File: packages/headless/src/ssr-commerce.index.ts [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+// Types & Helpers
+export {buildSSRSearchParameterSerializer} from './features/search-parameters/search-parameter-serializer.ssr';
+export type {
+  BaseProduct,
+  Product,
+  ChildProduct,
+} from './api/commerce/common/product';
+export type {FieldDescription} from './api/search/fields/fields-response';
+export type {Raw} from './api/search/search/raw';
+export type {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:146]: I don't think we need to export any of these

File: packages/headless/src/ssr-commerce.index.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+  SortOrder,
+  buildDateSortCriterion,
+  buildCriterionExpression,
+  buildFieldSortCriterion,
+  buildNoSortCriterion,
+  buildQueryRankingExpressionSortCriterion,
+  buildRelevanceSortCriterion,
+} from './features/sort-criteria/criteria';
+export {parseCriterionExpression} from './features/sort-criteria/criteria-parser';
+export type {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:169]: We should instead export from product-templates-manager

File: packages/headless/src/ssr-commerce.index.ts [hunk 7]
```diff
@@ -0,0 +1,222 @@
...
+  RangeFacetSortCriterion,
+  RangeFacetRangeAlgorithm,
+} from './features/facets/range-facets/generic/interfaces/request';
+export {
+  MinimumFieldsToInclude,
+  DefaultFieldsToInclude,
+  EcommerceDefaultFieldsToInclude,
+} from './features/fields/fields-state';
+export {buildSearchParameterSerializer} from './features/search-parameters/search-parameter-serializer';
+export type {FunctionExecutionTrigger} from './features/triggers/triggers-state';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:209]: We don't have support for execute triggers in headless commerce ATM

File: packages/headless/src/ssr-commerce.index.ts [hunk 8]
```diff
@@ -0,0 +1,222 @@
...
+  InferStaticState,
+  InferHydratedState,
+  InferBuildResult,
+} from './app/ssr-engine/types/core-engine';
+export type {LoggerOptions} from './app/logger';
+
+export type {LogLevel} from './app/logger';
+
+// State
+export type {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:53]: We should export from commerce-app-state instead

File: packages/headless/src/ssr-commerce.index.ts [hunk 9]
```diff
@@ -0,0 +1,222 @@
+import {polyfillCryptoNode} from './api/analytics/analytics-crypto-polyfill';
+
+polyfillCryptoNode();
+
+// 3rd Party Libraries
+export type {Unsubscribe, Middleware} from '@reduxjs/toolkit';
+export {createAction, createAsyncThunk, createReducer} from '@reduxjs/toolkit';
+export type {AnalyticsClientSendEventHook} from 'coveo.analytics';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:8]: We don't need this in commerce

File: packages/headless/src/ssr-commerce.index.ts [hunk 10]
```diff
@@ -0,0 +1,222 @@
...
+export type {
+  SSRCommerceEngine as CommerceEngine,
+  CommerceEngineDefinition,
+  CommerceEngineDefinitionOptions,
+} from './app/commerce-engine/commerce-engine.ssr';
+export {defineCommerceEngine} from './app/commerce-engine/commerce-engine.ssr';
+export {getSampleCommerceEngineConfiguration} from './app/commerce-engine/commerce-engine-configuration';
+
+// export type
+export type {CoreEngine, ExternalEngineOptions} from './app/engine';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:25]: I think we'd want CoreEngineNext, not CoreEngine

File: packages/headless/src/ssr-commerce.index.ts [hunk 11]
```diff
@@ -0,0 +1,222 @@
...
+export type {
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:26]: We should exporte CommerceEngineConfiguration from .app/commerce-engine/commerce-engine-configuration

File: packages/headless/src/ssr-commerce.index.ts [hunk 12]
```diff
@@ -0,0 +1,222 @@
...
+  facetRequestSelector,
+  facetResponseSelector,
+  facetResponseSelectedValuesSelector,
+} from './features/facets/facet-set/facet-set-selectors';
+
+export {
+  currentPageSelector,
+  maxPageSelector,
+  currentPagesSelector,
+} from './features/pagination/pagination-selectors';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:92]: Same comment as above, for now I don't think we should export any selectors unless we're certain they're gonna be useful

File: packages/headless/src/ssr-commerce.index.ts [hunk 13]
```diff
@@ -0,0 +1,222 @@
...
+export type {AnyFacetValueRequest} from './features/facets/generic/interfaces/generic-facet-request';
+export type {
+  RangeFacetSortCriterion,
+  RangeFacetRangeAlgorithm,
+} from './features/facets/range-facets/generic/interfaces/request';
+export {
+  MinimumFieldsToInclude,
+  DefaultFieldsToInclude,
+  EcommerceDefaultFieldsToInclude,
+} from './features/fields/fields-state';
```
Comment (fbeaudoincoveo) [inline packages/headless/src/ssr-commerce.index.ts:207]: I'd remove all of these exports

**Review by fbeaudoincoveo — APPROVED** [review]

**Review by fpbrault — APPROVED** [review]

**Review by alexprudhomme — APPROVED** [review]
Looks good but I agree with fred. Best to keep it very lean when starting. Make sure we only export the necessary stuff.

**Pull request merged**